MoltenVK/MoltenVK/Commands/MVKCommandEncoderState.h

```diff
@@ -384,17 +385,18 @@ class MVKResourcesCommandEncoderState : public MVKCommandEncoderState {
 
         if ( !b.mtlResource ) { return; }
 
-        T db = b;   // Copy that can be marked dirty
         MVKCommandEncoderState::markDirty();
         bindingsDirtyFlag = true;
-        db.isDirty = true;
 
         for (auto iter = bindings.begin(), end = bindings.end(); iter != end; ++iter) {
-            if( iter->index == db.index ) {
-                *iter = db;
-                return;
-            }
+            if (iter->index != b.index) continue;
```

Comment:
Why is this flow not the more direct positive form:

```
if (iter->index == b.index) {
    iter->update(b);
    return;
}
```

Reply:
I guess I slightly more like it my way, but I'll change to this next time I update the branch.